ext/liquid_c/lexer.c

```diff
@@ -144,7 +144,38 @@ const char *lex_one(const char *start, const char *end, lexer_token_t *token)
 
     if (is_special(c)) RETURN_TOKEN(c, 1);
 
-    rb_enc_raise(utf8_encoding, cLiquidSyntaxError, "Unexpected character %c", c);
+    int remaining_str_len = end - str;
```

Comment:
To fix this compiler warning:

```
lexer.c:147:33: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
    int remaining_str_len = end - str;
        ~~~~~~~~~~~~~~~~~   ~~~~^~~~~
```

```suggestion
    long remaining_str_len = end - str;
```